src/HttpClientInterception/HttpClientInterceptorOptions.cs

```diff
@@ -187,8 +191,16 @@ public HttpClientInterceptorOptions Deregister(HttpMethod method, Uri uri)
                 StatusCode = statusCode
             };
 
-            string key = BuildKey(method, uri);
-            _mappings[key] = interceptor;
+            if (ignoreQueryString)
+            {
+                string key = BuildKeyIgnoringQueryString(method, uri);
+                _ignoreQueryStringMappings[key] = interceptor;
```

Comment:
I wonder if there's a way we can handle this without requiring two separate dictionaries, like some sort of key prefix?

Reply:
Yep OK that would work.